apps/email/js/cards/cmp/autocomplete_source.js

```diff
@@ -0,0 +1,48 @@
+'use strict';
+define(function(require) {
+
+var contacts = navigator.mozContacts;
+
+function makeResult(query, contacts) {
+  return {
+    query: query,
+    contacts: contacts
+  };
+}
+
+/**
+ * Finds matches for the given query in the name and email for known people.
+ * Returns a promise that resolves to an object that has the query and a
+ * "match" array of mozContact entries. The match array will be a zero length
+ * array for no matches. See this bug about expanding the data source beyond
+ * contacts matches:
+ * https://bugzilla.mozilla.org/show_bug.cgi?id=1188710
+ * @param  {String} query the name/email query to use.
+ * @return {Promise}
+ */
+return function match(query) {
+  return new Promise(function(resolve, reject) {
+    if (!contacts || !query) {
+      return resolve(makeResult(query, []));
+    }
+
+    var request = contacts.find({
+      filterBy: ['name', 'email'],
+      filterOp: 'startsWith',
+      filterValue: query
+    });
```

Comment:
We probably also want to specify filterLimit with a value high enough that we'd expect the user to give up scrolling before they got there and instead type more, but not so high that we waste a lot of disk I/O time, IPC overhead, and then build a crazy big DOM.  Like 30 or 50 or something.  (What's the worst-case count on number of people someone would know named John/James/etc.? :)